<img src="https://raw.githubusercontent.com/carubbi/MQ/main/notebooks/assets/imgs/unifor-logo.png" width="400">
<br>
<b>
<font size="6" face="arial" color="blue">
    Graduação em Ciência da Computação
</font>
</b>
<br>
<b>
<font size="4" face="arial">
    Disciplina: Métodos Quantitativos em Computação
</font>
</b>

**Orientador: Prof. Me. Ricardo Carubbi** <br>
*Docente da Graduação e Pós-Graduação em Ciência de Dados e Inteligência Artificial*<br>
*Laboratório de Ciência de Dados e Inteligência Artificial*<br>
*Universidade de Fortaleza*<br>

# **Aula 2 — Fundamentos estatísticos e investigação com dados**



## 1. Objetivos de aprendizagem

Ao final da aula, o aluno será capaz de:

- formular uma pergunta estatística compatível com os dados disponíveis;
- distinguir população de interesse, população finita de referência, unidade de análise, unidade observacional e amostra;
- comparar amostras aleatórias e reconhecer a variabilidade amostral;
- avaliar se um procedimento de seleção é adequado ao objetivo da investigação;
- diferenciar variabilidade amostral de viés de seleção;
- distinguir descrição, inferência e afirmação causal;
- delimitar a quais unidades e populações as conclusões se aplicam, com base na origem e na seleção dos dados.



## 2. Agenda

1. Fundamentos de Estatística — 15 min
2. Pergunta de investigação e dados — 15 min
3. Investigação estatística e variabilidade — 15 min
4. População, amostra, representatividade e vieses — 35 min
5. Descrição, inferência e limites das conclusões — 20 min

**Código apresentado pronto:** importação do pandas, carregamento do arquivo, seleção das colunas e definição da população finita de referência. As demais operações serão desenvolvidas e interpretadas com a turma.

## 3. Fundamentos de Estatística


### 3.1 Estatística e variabilidade

A **Estatística** reúne técnicas para coletar, organizar, descrever, analisar e interpretar dados provenientes de observações ou experimentos, contribuindo para resolver problemas e produzir conhecimento (BARBETTA; BORNIA; REIS, 2010, seção 1.1).

A **variabilidade** corresponde às diferenças encontradas entre pessoas, animais, objetos ou situações estudadas. Pinguins, por exemplo, podem apresentar massas corporais diferentes. A Estatística considera essas diferenças para compreender fenômenos e apoiar decisões (BARBETTA; BORNIA; REIS, seções 1.1–1.2).

![Ciclo da pesquisa: dados, informações e novos conhecimentos e problemas.](../assets/imgs/u1_a02/figura_1_1_ciclo_pesquisa.png)

**Figura 1 - Processo iterativo das pesquisas empíricas.** Os resultados de uma pesquisa podem originar novas perguntas. Fonte: reprodução de Barbetta, Bornia e Reis (2010, figura 1.1, p. 12; página 13 do PDF).

### 3.2 Unidades, variáveis e dados

Para descrever um conjunto de dados, distinguiremos:

- **Unidade de análise:** entidade sobre a qual se deseja formular uma descrição ou conclusão, como uma pessoa, um animal ou uma escola.
- **Unidade observacional:** entidade na qual as características são observadas ou medidas.
- **Variável:** característica investigada, como espécie ou massa corporal.
- **Dado estatístico:** valor registrado de uma variável, como uma massa de 4.200 g.
- **Observação:** conjunto do que foi observado em um caso; em uma matriz de dados, corresponde a uma linha, enquanto cada coluna corresponde a uma variável.

Essa organização permite reunir vários dados de um mesmo caso, como espécie, ilha e massa corporal (BARBETTA; BORNIA; REIS, 2010, seção 3.1).

A **informação** é extraída dos dados por meio de sua análise e interpretação em relação ao problema investigado. Organizar e examinar os dados ajuda a compreender o fenômeno e a identificar aspectos relevantes para a pergunta (PINHEIRO et al., 2009, seção 1.1).

Unidade de análise e unidade observacional podem coincidir ou diferir. Se medimos pinguins e concluímos sobre pinguins, coincidem. Se coletamos respostas de alunos para concluir sobre escolas, os alunos são as unidades observacionais e as escolas são as unidades de análise (MORDEN, 2024, seção 3.7).

![Matriz de dados com indicações de indivíduos ou casos, variáveis e dados.](../assets/imgs/u1_a02/esquema_matriz_dados_barbetta.png)

**Figura 2 - Casos, variáveis e dados em uma matriz.** As linhas reúnem os dados de cada caso e as colunas correspondem às características registradas. Fonte: reprodução de Barbetta, Bornia e Reis (2010, seção 3.1, esquema sem numeração, p. 51; página 52 do PDF).

### 3.3 População e amostra

A **população de interesse** é o conjunto completo de elementos cujas características se deseja estudar. Uma **amostra** é um subconjunto desses elementos, selecionado segundo regras definidas (PINHEIRO et al., 2009, seção 1.2).

Se a pergunta trata dos 344 registros disponíveis do Palmer Penguins, esse conjunto é a população de interesse; dez registros selecionados formam uma amostra. Se a pergunta muda para todos os pinguins dessas espécies, a população de interesse passa a ser mais ampla que o conjunto disponível.

Por isso, é necessário definir a população antes de selecionar a amostra.

### 3.4 Estatística Descritiva e Inferência

A **Estatística Descritiva** organiza, resume e apresenta os dados observados, por meio de tabelas, gráficos e resumos numéricos (BARBETTA; BORNIA; REIS, 2010, abertura do capítulo 3).

A **Inferência Estatística** utiliza resultados de uma amostra para formular conclusões sobre a população da qual ela foi extraída (BARBETTA; BORNIA; REIS, 2010, seção 1.1). Essa passagem envolve incerteza: outras amostras podem produzir resultados diferentes. Sua validade depende de como os dados foram obtidos e das condições exigidas pelo método utilizado.

**Exemplo:** contar os pinguins de cada espécie entre dez registros selecionados descreve essa amostra. Usar esses resultados para responder sobre os 344 registros envolve inferência. O cálculo feito na amostra, por si só, não justifica a extensão da conclusão à população.

## 4. Pergunta de investigação e dados

> Como a massa corporal varia entre os pinguins registrados?

Os pinguins registrados têm massas corporais diferentes. Vamos examinar essas diferenças e discutir quais perguntas os dados permitem responder. Primeiro, conheceremos as etapas da investigação e o conjunto de dados utilizado.

### 4.1 Etapas da investigação

Uma investigação estatística articula etapas dependentes entre si:

1. delimitar o problema;
2. formular uma pergunta compatível com os dados;
3. definir a população de interesse, a unidade de análise e a unidade observacional;
4. examinar ou planejar a obtenção dos dados;
5. analisar a variabilidade e os padrões;
6. comunicar a conclusão e explicitar seus limites.

Uma falha nas primeiras etapas não é corrigida automaticamente por cálculos corretos nas etapas finais (BARBETTA; BORNIA; REIS, 2010, seção 2.1).

![Etapas da pesquisa, da definição do problema às conclusões, relacionadas às metodologias da área de estudo e estatística.](../assets/imgs/u1_a02/figura_2_1_processo_investigacao.png)

**Figura 3 - Etapas usuais de uma pesquisa empírica.** A metodologia estatística participa do planejamento, da execução e da análise da pesquisa. Fonte: reprodução de Barbetta, Bornia e Reis (2010, figura 2.1, p. 23; página 24 do PDF).

### 4.2 Dados e preparação

A **coleta de dados** é a obtenção e o registro das informações necessárias à investigação. **Dados primários** são coletados diretamente para responder à pergunta da investigação. **Dados secundários** foram obtidos anteriormente para outra finalidade e são reutilizados no estudo atual (STATISTICS CANADA, 2010, p. 6–7).

Utilizaremos 344 registros do Palmer Penguins, com informações sobre espécie, ilha e massa corporal. Para nossa atividade, são dados secundários: utilizamos uma base produzida por outros pesquisadores. A unidade de análise e a unidade observacional coincidem neste exemplo: observamos características dos pinguins e formulamos conclusões sobre eles.

A **população finita de referência** é o conjunto delimitado adotado como base para selecionar e comparar amostras nesta atividade. Aqui, corresponde aos 344 registros disponíveis. As amostras serão subconjuntos desses registros.

Adotar esse conjunto como referência não o torna uma representação automática de todos os pinguins das três espécies (HORST; HILL; GORMAN, 2020).

In [1]:
# Importar a biblioteca utilizada na aula.
import pandas as pd


In [2]:
# Definir a localização e carregar os dados.
arquivo_dados = "https://raw.githubusercontent.com/carubbi/MQ/main/data/raw/penguins_raw.csv"
df = pd.read_csv(arquivo_dados)

# Visualizar as primeiras observações.
df.head()


,studyName,Sample Number,Species,Region,Island,Stage,Individual ID,Clutch Completion,Date Egg,Culmen Length (mm),Culmen Depth (mm),Flipper Length (mm),Body Mass (g),Sex,Delta 15 N (o/oo),Delta 13 C (o/oo),Comments
0,PAL0708,1,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A1,Yes,2007-11-11,39.1,18.7,181.0,3750.0,MALE,NaN,NaN,Not enough blood for isotopes.
1,PAL0708,2,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N1A2,Yes,2007-11-11,39.5,17.4,186.0,3800.0,FEMALE,8.94956,-24.69454,NaN
2,PAL0708,3,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A1,Yes,2007-11-16,40.3,18.0,195.0,3250.0,FEMALE,8.36821,-25.33302,NaN
3,PAL0708,4,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N2A2,Yes,2007-11-16,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Adult not sampled.
4,PAL0708,5,Adelie Penguin (Pygoscelis adeliae),Anvers,Torgersen,"Adult, 1 Egg Stage",N3A1,Yes,2007-11-16,36.7,19.3,193.0,3450.0,FEMALE,8.76651,-25.32426,NaN


**Tabela 1 - Primeiras observações do conjunto de dados.** Cada linha registra um pinguim e cada coluna armazena uma característica ou informação do estudo. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Um **identificador** é um código usado para distinguir entidades ou registros. Para a atividade, serão mantidas as seguintes colunas:

- `studyName`: estudo em que a observação foi registrada;
- `Individual ID`: identificador do indivíduo dentro do estudo;
- `Species`: espécie do pinguim;
- `Island`: ilha onde o pinguim foi observado;
- `Body Mass (g)`: massa corporal, em gramas.

Como o identificador pode se repetir entre estudos, cada registro é identificado pela combinação de `studyName` + `Individual ID` (HORST; HILL; GORMAN, 2020).


In [3]:
# Escolher as colunas necessárias.
colunas_amostras = [
    "studyName",
    "Individual ID",
    "Species",
    "Island",
    "Body Mass (g)",
]

# Criar a população finita de referência.
populacao_referencia = df[colunas_amostras].copy()

# Verificar o número de linhas e colunas.
populacao_referencia.shape


(344, 5)

A saída `(344, 5)` informa que a população finita de referência possui 344 linhas e 5 colunas selecionadas.

### 4.3 Quais perguntas os dados permitem responder?

O **desenho do estudo** é a forma como a investigação é organizada, incluindo a seleção dos casos e a obtenção dos dados. Em um **estudo observacional**, os pesquisadores registram características e condições existentes, sem determinar qual condição cada unidade receberá. A **atribuição experimental** ocorre quando os pesquisadores definem qual condição ou tratamento cada unidade receberá para comparar os resultados.

| Pergunta | Avaliação | Justificativa |
| --- | --- | --- |
| Quantos pinguins foram registrados por espécie? | Compatível com os dados | A espécie está registrada no arquivo. |
| Como a massa corporal varia entre os registros com essa medida preenchida? | Compatível com os dados | A massa corporal está registrada para a maioria das observações. |
| A massa de cada pinguim mudou ao longo da vida? | Incompatível com os dados | O arquivo não acompanha repetidamente cada animal ao longo da vida. |
| A ilha causou diferenças de massa corporal? | Incompatível com o desenho do estudo | A ilha foi observada, sem atribuição experimental; a comparação, por si só, não permite separar seu efeito de outras diferenças entre os pinguins. |

## 5. Investigação estatística e variabilidade

No Palmer Penguins, a variabilidade aparece nas diferenças de espécie, ilha e massa corporal entre os registros. Examinaremos essas diferenças antes de selecionar amostras. Elas podem refletir características biológicas, contexto, período, medição ou preparação dos dados.

Um valor incomum não deve ser classificado automaticamente como erro. Primeiro, deve-se verificar a unidade de medida, o procedimento de coleta, o registro original e a coerência com as demais variáveis (BARBETTA; BORNIA; REIS, seção 1.2; MONTGOMERY; RUNGER, seção 1.1.1).

### 5.1 Reconhecer a composição e a variabilidade dos registros

Antes de selecionar amostras, examinaremos a composição da população finita de referência e a variação da massa corporal. Esses resultados servirão de comparação para as seleções posteriores.

In [4]:
# Contar quantas observações existem de cada espécie.
composicao_referencia = populacao_referencia["Species"].value_counts()
composicao_referencia


Species
Adelie Penguin (Pygoscelis adeliae)          152
Gentoo penguin (Pygoscelis papua)            124
Chinstrap penguin (Pygoscelis antarctica)     68
Name: count, dtype: int64

**Tabela 2 - Composição da população de referência por espécie.** A tabela será a referência de comparação para as amostras produzidas nesta aula. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [5]:
# Contar quantas observações existem em cada ilha.
composicao_ilhas = populacao_referencia["Island"].value_counts()
composicao_ilhas


Island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

**Tabela 3 - Composição da população de referência por ilha.** As quantidades mostram que as ilhas não aparecem com a mesma frequência no arquivo. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Antes de sortear, a equipe conhece a composição do conjunto disponível. Isso permite verificar depois quais grupos apareceram ou ficaram ausentes das amostras e quais apresentaram proporção maior na amostra do que na população de referência.

A **média aritmética** é a soma dos valores dividida pela quantidade de valores utilizados. Para a massa corporal, usaremos apenas os registros com essa medida preenchida. No resumo produzido por `describe()`, `count` indica essa quantidade, `mean` indica a média, e `min` e `max` indicam o menor e o maior valor.

In [6]:
# Resumir a massa corporal para reconhecer sua variabilidade.
resumo_massa = populacao_referencia["Body Mass (g)"].describe()
resumo_massa


count     342.000000
mean     4201.754386
std       801.954536
min      2700.000000
25%      3550.000000
50%      4050.000000
75%      4750.000000
max      6300.000000
Name: Body Mass (g), dtype: float64

**Tabela 4 - Resumo da massa corporal.** A saída descreve os valores disponíveis de `Body Mass (g)` e permite reconhecer o número de registros com massa corporal informada, posição e variabilidade. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** formule uma pergunta compatível com os dados sobre a massa corporal e explique por que um valor incomum não deve ser classificado automaticamente como erro.

**Resposta:** uma pergunta compatível com os dados é: “Como varia a massa corporal entre os pinguins registrados no arquivo?”. Há 342 valores não ausentes, entre 2.700 g e 6.300 g. Um valor incomum pode refletir variação biológica legítima; antes de classificá-lo como erro, é necessário verificar a unidade, o procedimento de medição, o registro original e sua coerência com as demais variáveis.

### 5.2 Variabilidade entre amostras

A **variabilidade amostral** ocorre quando amostras diferentes, obtidas pelo mesmo procedimento, produzem resultados diferentes. Duas amostras de dez registros, por exemplo, podem conter quantidades distintas de cada espécie e apresentar massas corporais médias diferentes.

Na próxima seção, definiremos o procedimento de seleção e compararemos duas amostras da população finita de referência. A comparação permitirá examinar diferenças entre seleções feitas pelo mesmo procedimento.

## 6. População, amostra, representatividade e vieses

A população de interesse depende da pergunta. Para descrever os registros das três espécies, consideramos o conjunto completo; para descrever somente Gentoo, consideramos esse subconjunto. Em cada caso, precisamos decidir quais unidades observar e como selecioná-las.

- **Censo:** levantamento de todos os elementos da população (PINHEIRO et al., 2009, seção 1.2).
- **Amostragem:** processo de selecionar os elementos que integrarão uma amostra (BARBETTA; BORNIA; REIS, 2010, seções 1.6 e 2.2).
- **Quadro amostral:** lista ou outra representação das unidades acessíveis à seleção.
- **Amostragem aleatória simples:** procedimento no qual cada amostra possível de mesmo tamanho tem a mesma probabilidade, isto é, a mesma chance, de ser selecionada.

Uma amostra obtida por esse sorteio é uma **amostra aleatória simples**. Nos exercícios, a seleção será **sem reposição**: um registro sorteado não pode ser escolhido novamente na mesma amostra.

Analisar todos os 344 registros descreve integralmente a população finita de referência. Isso não equivale a realizar um censo de todos os pinguins dessas espécies. Mesmo um censo pode conter erros de registro, medição ou omissão de unidades; observar toda a população não garante ausência de erros.

Nesta aula, as linhas disponíveis constituem o quadro amostral. O sorteio distribui as chances entre essas linhas; pinguins ausentes do arquivo não podem ser selecionados (NAVIDI, seção 1.1; PINHEIRO et al., seção 1.2).

![Relação entre população, amostragem, amostra e inferência.](../assets/imgs/u1_a02/figura_2_2_populacao_amostra_inferencia.png)

**Figura 4 - Relação entre população, amostra, amostragem e inferência.** A inferência parte da amostra para responder sobre uma população definida, mas sua validade depende do procedimento de seleção e das condições assumidas. Fonte: reprodução de Barbetta, Bornia e Reis (2010, figura 2.2, p. 26; página 27 do PDF).

### 6.1 Representatividade não é sinônimo de tamanho

A **cobertura** diz respeito a quais unidades da população de interesse estão incluídas no quadro amostral. A **representatividade** expressa a adequação da amostra para responder sobre essa população em relação às características estudadas; sua avaliação exige examinar a cobertura e o procedimento de seleção.

Uma amostra aleatória não precisa reproduzir exatamente a composição da população. Aumentar seu tamanho pode reduzir a variabilidade amostral, mas não corrige automaticamente a exclusão sistemática de grupos.

No Palmer Penguins, sortear mais linhas não inclui animais que ficaram fora da coleta original. Portanto, um sorteio dentro do arquivo não garante representatividade de uma população biológica mais ampla (BARBETTA; BORNIA; REIS, 2010, seção 2.2.1; NAVIDI, 2024, seção 1.1).

### 6.2 Variabilidade e viés não são a mesma coisa

Um **viés** é uma tendência sistemática de um procedimento a produzir resultados que se afastam da característica que se deseja conhecer. Nesta aula, distinguiremos problemas de seleção e de medição das diferenças ocasionais entre amostras.

| Fenômeno | Característica | Exemplo nesta aula |
| --- | --- | --- |
| Variabilidade amostral | Resultados mudam entre amostras obtidas pelo mesmo procedimento. | Duas amostras aleatórias de dez registros apresentam composições diferentes. |
| Viés de seleção | O procedimento favorece ou exclui sistematicamente unidades relevantes. | Filtrar somente Gentoo quando o objetivo é comparar as três espécies. |
| Viés de medição | O registro se afasta sistematicamente da característica que se pretende medir. | Deve ser considerado conceitualmente, mas não pode ser diagnosticado apenas com as saídas usadas nesta aula. |

A aleatoriedade aplicada depois de um filtro inadequado não recupera os grupos já excluídos. Por outro lado, diferenças ocasionais entre duas amostras aleatórias não bastam para concluir que houve viés.

### 6.3 Comparar amostras aleatórias reproduzíveis

Duas amostras de mesmo tamanho serão obtidas por amostragem aleatória simples sem reposição. No código, a **semente** é um valor que permite reproduzir a sequência usada pelo programa no sorteio; ela será indicada por `random_state`. Usaremos sementes diferentes para obter duas seleções.

A comparação permitirá examinar diferenças entre sorteios feitos pelo mesmo procedimento. A possibilidade de reproduzir um sorteio não garante a representatividade da amostra.

In [7]:
# Sortear 10 observações sem reposição.
amostra_a = populacao_referencia.sample(
    n=10,
    random_state=42,
)

amostra_a


,studyName,Individual ID,Species,Island,Body Mass (g)
194,PAL0809,N7A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
157,PAL0708,N33A2,Gentoo penguin (Pygoscelis papua),Biscoe,4550.0
225,PAL0809,N55A2,Gentoo penguin (Pygoscelis papua),Biscoe,5200.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
318,PAL0809,N75A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3550.0
329,PAL0910,N92A2,Chinstrap penguin (Pygoscelis antarctica),Dream,4050.0
319,PAL0809,N75A2,Chinstrap penguin (Pygoscelis antarctica),Dream,3500.0
260,PAL0910,N29A1,Gentoo penguin (Pygoscelis papua),Biscoe,4575.0
114,PAL0910,N61A1,Adelie Penguin (Pygoscelis adeliae),Biscoe,3900.0
220,PAL0809,N53A1,Gentoo penguin (Pygoscelis papua),Biscoe,4700.0


**Tabela 5 - Amostra aleatória A.** Foram sorteadas 10 linhas da população finita de referência. O valor de `random_state` permite reproduzir o mesmo resultado (PANDAS DEVELOPMENT TEAM, 2026). Fonte dos dados: Horst, Hill e Gorman (2020).

In [8]:
# Repetir o sorteio com outra semente.
amostra_b = populacao_referencia.sample(
    n=10,
    random_state=7,
)

amostra_b


,studyName,Individual ID,Species,Island,Body Mass (g)
173,PAL0708,N41A2,Gentoo penguin (Pygoscelis papua),Biscoe,5000.0
30,PAL0708,N21A1,Adelie Penguin (Pygoscelis adeliae),Dream,3250.0
115,PAL0910,N61A2,Adelie Penguin (Pygoscelis adeliae),Biscoe,4075.0
341,PAL0910,N99A2,Chinstrap penguin (Pygoscelis antarctica),Dream,3775.0
141,PAL0910,N80A2,Adelie Penguin (Pygoscelis adeliae),Dream,3475.0
73,PAL0809,N35A2,Adelie Penguin (Pygoscelis adeliae),Torgersen,4150.0
336,PAL0910,N96A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3950.0
328,PAL0910,N92A1,Chinstrap penguin (Pygoscelis antarctica),Dream,3600.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
28,PAL0708,N18A1,Adelie Penguin (Pygoscelis adeliae),Biscoe,3150.0


**Tabela 6 - Amostra aleatória B.** A mudança de `random_state` produz outra seleção reproduzível de 10 linhas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [9]:
# Contar as espécies presentes nas duas amostras.
composicao_amostra_a = amostra_a["Species"].value_counts()
composicao_amostra_b = amostra_b["Species"].value_counts()

# Reunir as contagens para facilitar a comparação.
comparacao_composicoes = pd.DataFrame(
    {
        "População de referência": composicao_referencia,
        "Amostra A": composicao_amostra_a,
        "Amostra B": composicao_amostra_b,
    }
).fillna(0).astype(int)

comparacao_composicoes


,População de referência,Amostra A,Amostra B
Species,,,
Adelie Penguin (Pygoscelis adeliae),152,1,5
Chinstrap penguin (Pygoscelis antarctica),68,3,3
Gentoo penguin (Pygoscelis papua),124,6,2


**Tabela 7 - Contagens por espécie na população e nas amostras.** As duas amostras foram obtidas pelo mesmo procedimento, mas apresentam contagens diferentes. Como a população possui 344 observações e cada amostra possui 10, a comparação proporcional será feita na próxima tabela. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).


A **proporção** de uma espécie é sua quantidade de registros dividida pelo total de registros do conjunto analisado. Multiplicar essa razão por 100 a expressa em **porcentagem**. Usaremos 344 como total na população de referência e 10 em cada amostra; assim, poderemos comparar conjuntos de tamanhos diferentes.

In [10]:
# Calcular as proporções de cada espécie, em porcentagem.
proporcoes_referencia = composicao_referencia / len(populacao_referencia) * 100
proporcoes_amostra_a = composicao_amostra_a / len(amostra_a) * 100
proporcoes_amostra_b = composicao_amostra_b / len(amostra_b) * 100

# Reunir e arredondar as porcentagens.
comparacao_proporcoes = pd.DataFrame(
    {
        "População de referência (%)": proporcoes_referencia,
        "Amostra A (%)": proporcoes_amostra_a,
        "Amostra B (%)": proporcoes_amostra_b,
    }
).fillna(0).round(1)

comparacao_proporcoes


,População de referência (%),Amostra A (%),Amostra B (%)
Species,,,
Adelie Penguin (Pygoscelis adeliae),44.2,10.0,50.0
Chinstrap penguin (Pygoscelis antarctica),19.8,30.0,30.0
Gentoo penguin (Pygoscelis papua),36.0,60.0,20.0


**Tabela 8 - Proporções das espécies na população e nas amostras.** Adelie corresponde a 44,2% da população de referência, mas representa 10,0% da amostra A e 50,0% da amostra B. Gentoo passa de 36,0% na população para 60,0% na amostra A e 20,0% na amostra B. Essas diferenças ilustram a variabilidade da composição entre amostras aleatórias pequenas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** a seleção aleatória produziu uma cópia exata da composição da população? Essas diferenças, isoladamente, comprovam viés?

**Resposta:** não. Nenhuma das amostras reproduziu exatamente a composição da população de referência. As diferenças observadas entre sorteios de dez registros são compatíveis com variabilidade amostral e, isoladamente, não comprovam viés. Para caracterizar viés, seria necessário identificar um mecanismo de seleção que favoreça ou exclua sistematicamente grupos relevantes ao objetivo.


In [11]:
# Calcular a média da massa corporal em cada conjunto.
media_massa_referencia = populacao_referencia["Body Mass (g)"].mean()
media_massa_amostra_a = amostra_a["Body Mass (g)"].mean()
media_massa_amostra_b = amostra_b["Body Mass (g)"].mean()

# Reunir e arredondar as médias.
comparacao_medias = pd.Series(
    {
        "População de referência": media_massa_referencia,
        "Amostra A": media_massa_amostra_a,
        "Amostra B": media_massa_amostra_b,
    },
    name="Média da massa corporal (g)",
).round(1)

comparacao_medias


População de referência    4201.8
Amostra A                  4262.5
Amostra B                  3872.5
Name: Média da massa corporal (g), dtype: float64

**Tabela 9 - Média da massa corporal na população e nas amostras.** A média é 4.201,8 g na população de referência, 4.262,5 g na amostra A e 3.872,5 g na amostra B. A mesma medida assume valores diferentes porque foi calculada em seleções diferentes. Isso evidencia variabilidade amostral e não demonstra, por si só, a existência de viés. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).


### 6.4 O objetivo determina a adequação do filtro

Considere dois objetivos:

- **Objetivo A:** comparar Adelie, Chinstrap e Gentoo. Um filtro que mantém somente Gentoo é inadequado porque impede a comparação pretendida.
- **Objetivo B:** descrever somente os registros Gentoo. O filtro seleciona os registros da população de interesse definida pela pergunta.

O código não é adequado ou inadequado isoladamente. Sua validade depende da correspondência entre pergunta, população e procedimento de seleção.

#### Objetivo A: descrever as três espécies

Suponha que a equipe queira descrever conjuntamente Adelie, Chinstrap e Gentoo, mas filtre apenas Gentoo antes do sorteio. Primeiro, o filtro será aplicado; depois, o subconjunto será amostrado.

In [12]:
# Criar uma condição para selecionar somente Gentoo.
filtro_gentoo = (
    populacao_referencia["Species"]
    == "Gentoo penguin (Pygoscelis papua)"
)

# Aplicar o filtro.
populacao_gentoo = populacao_referencia[filtro_gentoo]

# Verificar o tamanho do subconjunto.
populacao_gentoo.shape


(124, 5)

In [13]:
# Sortear 10 observações dentro do subconjunto Gentoo.
amostra_gentoo = populacao_gentoo.sample(
    n=10,
    random_state=42,
)

amostra_gentoo


,studyName,Individual ID,Species,Island,Body Mass (g)
170,PAL0708,N40A1,Gentoo penguin (Pygoscelis papua),Biscoe,4800.0
194,PAL0809,N7A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
188,PAL0809,N4A1,Gentoo penguin (Pygoscelis papua),Biscoe,4950.0
228,PAL0809,N58A1,Gentoo penguin (Pygoscelis papua),Biscoe,4600.0
205,PAL0809,N14A2,Gentoo penguin (Pygoscelis papua),Biscoe,5000.0
242,PAL0910,N15A1,Gentoo penguin (Pygoscelis papua),Biscoe,4950.0
208,PAL0809,N16A1,Gentoo penguin (Pygoscelis papua),Biscoe,4300.0
274,PAL0910,N43A1,Gentoo penguin (Pygoscelis papua),Biscoe,5200.0
273,PAL0910,N39A2,Gentoo penguin (Pygoscelis papua),Biscoe,5750.0
156,PAL0708,N33A1,Gentoo penguin (Pygoscelis papua),Biscoe,5400.0


**Tabela 10 - Amostra obtida após o filtro por espécie.** O sorteio é aleatório dentro do subconjunto Gentoo, mas Adelie e Chinstrap não podem ser selecionadas. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

In [14]:
# Verificar a relação observada entre espécie e ilha.
especies_por_ilha = pd.crosstab(
    populacao_referencia["Species"],
    populacao_referencia["Island"],
)

especies_por_ilha


Island,Biscoe,Dream,Torgersen
Species,,,
Adelie Penguin (Pygoscelis adeliae),44,56,52
Chinstrap penguin (Pygoscelis antarctica),0,68,0
Gentoo penguin (Pygoscelis papua),124,0,0


**Tabela 11 - Espécies observadas por ilha.** No conjunto de dados, Gentoo aparece somente em Biscoe; Chinstrap, somente em Dream; e Adelie, nas três ilhas. Assim, filtrar uma ilha também pode alterar quais espécies podem integrar a seleção. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

Para o **Objetivo A**, o procedimento que produziu a amostra Gentoo apresenta viés de seleção: o procedimento exclui grupos que pertencem à população definida pelo objetivo. A aleatoriedade aplicada depois do filtro não corrige essa exclusão (BARBETTA; BORNIA; REIS, 2010, seção 2.2.1; NAVIDI, 2024, seção 1.1).


#### Objetivo B: descrever somente Gentoo

Se a pergunta mudar para “como são os pinguins Gentoo registrados no conjunto de dados?”, o filtro deixa de ser um erro. A população de interesse passa a ser o subconjunto Gentoo, e a amostra é sorteada dentro do grupo adequado.

O mesmo código pode ser coerente ou inadequado conforme o objetivo da investigação. Por isso, a população de interesse deve ser definida **antes** da seleção dos dados.


### 6.5 Aplicação orientada

Considere a seguinte solicitação:

> A equipe deseja comparar as três espécies, mas decidiu coletar uma amostra apenas na ilha Biscoe.

Execute a próxima célula e registre, em uma célula Markdown:

1. Qual é a população de interesse declarada?
2. Quais espécies podem aparecer na seleção feita em Biscoe?
3. Qual espécie é excluída e qual fica restrita?
4. A amostra permite comparar as três espécies? Justifique com a saída produzida.


In [15]:
# Selecionar as observações da ilha Biscoe.
filtro_biscoe = populacao_referencia["Island"] == "Biscoe"
populacao_biscoe = populacao_referencia[filtro_biscoe]

# Contar as espécies disponíveis após o filtro.
especies_em_biscoe = populacao_biscoe["Species"].value_counts()
especies_em_biscoe


Species
Gentoo penguin (Pygoscelis papua)      124
Adelie Penguin (Pygoscelis adeliae)     44
Name: count, dtype: int64

**Tabela 12 - Espécies disponíveis após o filtro por Biscoe.** A saída fornece a evidência necessária para avaliar se o procedimento atende ao objetivo de comparar as três espécies. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Resposta:** a população de interesse declarada reúne os pinguins registrados das três espécies. Após o filtro por Biscoe, somente Gentoo e Adelie podem aparecer: Chinstrap é excluída e Adelie fica restrita às 44 observações dessa ilha. Portanto, a seleção não permite comparar adequadamente as três espécies e apresenta viés em relação ao objetivo declarado.


## 7. Descrição, inferência e limites das conclusões

| Tipo de afirmação | O que faz | Exemplo |
| --- | --- | --- |
| Descritiva | Resume as observações disponíveis. | Informar quantos registros de cada espécie aparecem no arquivo. |
| Inferencial | Usa uma amostra para responder sobre uma população definida, considerando a incerteza e as condições do método. | Estimar a proporção de Adelie em uma população-alvo a partir de uma amostra representativa. |
| Causal | Afirma que uma condição ou ação produz uma mudança no resultado. | Afirmar que a ilha causou determinada massa corporal. |

Os cálculos deste notebook descrevem corretamente o arquivo e as amostras sorteadas dentro dele. Eles não sustentam automaticamente inferências sobre todos os pinguins das espécies nem afirmações causais sobre ilha, espécie ou massa corporal.

### 7.1 Parâmetro, estatística amostral, estimador e estimativa

Um **parâmetro** $\theta$ descreve uma característica numérica da população. Uma **estatística amostral** é uma medida calculada na amostra, como sua média ou uma proporção (BARBETTA; BORNIA; REIS, 2010, seção 1.6).

Uma **variável aleatória** associa um valor numérico ao resultado de um processo sujeito ao acaso. Aqui, $X_i$ representa o valor que poderá ser observado na posição $i$ da amostra antes do sorteio; $x_i$ é o valor efetivamente observado e $n$ é o tamanho da amostra. A notação $T$ representa a regra de cálculo aplicada aos valores da amostra.

Um **estimador** é uma regra de cálculo utilizada para estimar um parâmetro a partir dos dados de uma amostra (PINHEIRO et al., 2009, seção 7.1):

$$
\widehat{\theta}=T(X_1,X_2,\ldots,X_n).
$$

A **estimativa** é o valor numérico obtido ao aplicar essa regra aos dados observados (PINHEIRO et al., 2009, seção 7.1):

$$
\hat{\theta}=T(x_1,x_2,\ldots,x_n).
$$

**Aplicação aos resultados anteriores:** a média de massa corporal dos registros preenchidos na população finita de referência é um parâmetro desse conjunto. A média calculada nos registros preenchidos de uma amostra é uma estatística amostral. Quando essa regra é usada para estimar a média da população, ela funciona como estimador; o número obtido é a estimativa. A interpretação se refere às massas disponíveis e exige considerar as ausências.

A correção do cálculo não garante a validade da generalização. Também são necessários uma população bem definida, um procedimento de seleção justificável e hipóteses compatíveis com o método inferencial (PINHEIRO et al., 2009, seção 7.1).

### 7.2 Como delimitar as conclusões

Uma conclusão responsável informa:

- quais observações foram analisadas;
- qual população foi definida;
- como as unidades foram selecionadas;
- quais grupos ou períodos ficaram fora da cobertura;
- se o resultado é descritivo, inferencial ou causal;
- quais limitações impedem uma generalização mais ampla.

No Palmer Penguins, uma formulação apropriada começa por “nos registros disponíveis” ou “na amostra selecionada”. A expressão “em todos os pinguins dessas espécies” exigiria justificativas que o arquivo, isoladamente, não fornece.

In [16]:
# Calcular as contagens e as proporções observadas de cada espécie.
contagens_especies = populacao_referencia["Species"].value_counts()
proporcoes_especies = (contagens_especies / len(populacao_referencia) * 100).round(1)

# Calcular a quantidade válida, o menor e o maior valor de massa corporal.
massa_corporal = populacao_referencia["Body Mass (g)"]
amplitude_massa = pd.Series(
    {
        "Quantidade válida": massa_corporal.count(),
        "Mínimo": massa_corporal.min(),
        "Máximo": massa_corporal.max(),
    }
)

# Reunir as descrições em uma única tabela.
descricao_arquivo = pd.concat(
    [contagens_especies, proporcoes_especies, amplitude_massa],
    keys=["Contagem por espécie", "Proporção por espécie (%)", "Massa corporal (g)"],
    names=["Medida", "Categoria"],
).to_frame(name="Valor")

descricao_arquivo


Valor
Medida                    Categoria                                        
Contagem por espécie      Adelie Penguin (Pygoscelis adeliae)         152.0
                          Gentoo penguin (Pygoscelis papua)           124.0
                          Chinstrap penguin (Pygoscelis antarctica)    68.0
Proporção por espécie (%) Adelie Penguin (Pygoscelis adeliae)          44.2
                          Gentoo penguin (Pygoscelis papua)            36.0
                          Chinstrap penguin (Pygoscelis antarctica)    19.8
Massa corporal (g)        Quantidade válida                           342.0
                          Mínimo                                     2700.0
                          Máximo                                     6300.0

**Tabela 13 - Descrição do arquivo por espécie e massa corporal.** As contagens, as proporções e os valores mínimo e máximo pertencem às observações disponíveis. Fonte: elaboração própria com dados de Horst, Hill e Gorman (2020).

**Registro do estudante:** escreva uma conclusão descritiva sustentada pela saída e uma generalização que a seleção disponível não permite sustentar.

**Resposta:** no arquivo, Adelie possui 152 registros (44,2%), Gentoo 124 (36,0%) e Chinstrap 68 (19,8%). Entre os 342 valores não ausentes de massa corporal, os valores variam de 2.700 g a 6.300 g. Esses resultados não permitem afirmar que as mesmas proporções ou os mesmos valores mínimo e máximo caracterizam todos os pinguins dessas espécies, pois o arquivo não é uma amostra probabilística dessa população biológica.

## 8. Estudo e exercícios



### Materiais didáticos

- **Barbetta, Bornia e Reis:** seções 1.1–1.2, 1.6 e 2.1–2.2.1; abertura do capítulo 3 e seção 3.1: Estatística, investigação, população, amostragem, descrição, dados e variáveis.
- **Morden:** seção 3.7, páginas 65–66: unidade de análise e unidade observacional.
- **Pinheiro et al.:** seções 1.1–1.2: análise dos dados, população, amostra e censo; seção 7.1, página 193: parâmetro, estimador e estimativa.
- **Statistics Canada:** *Teacher’s Guide to Data Discovery*, páginas 6–7: dados primários e secundários.
- **Palmer Penguins:** documentação da fonte, das variáveis e da coleta.



### Exercícios indicados

- Banco de questões e provas 2026.2, questões 5 e 6, página 8: investigação estatística, população e amostra.
- Barbetta, Bornia e Reis, exercício 1.2, página 23: população, amostra e amostragem.
- Barbetta, Bornia e Reis, exercício 2.7, página 34: amostragem e representatividade.

## 9. Referências

1. BARBETTA, Pedro Alberto; BORNIA, Antonio Cezar; REIS, Marcelo Menezes. **Estatística para cursos de engenharia e informática**. 3. ed. São Paulo: Atlas, 2010.

2. HORST, Allison Marie; HILL, Alison Presmanes; GORMAN, Kristen B. **palmerpenguins: Palmer Archipelago (Antarctica) penguin data**. 2020. DOI: 10.5281/zenodo.3960218.

3. MONTGOMERY, Douglas C.; RUNGER, George C. **Applied Statistics and Probability for Engineers**. 7. ed. Hoboken: Wiley, 2018.

4. MORDEN, Peter. **Quantitative Research Methods for the Applied Human Sciences**. 2024.

5. NAVIDI, William. **Statistics for Engineers and Scientists**. 6. ed. New York: McGraw Hill, 2024.

6. PANDAS DEVELOPMENT TEAM. **pandas documentation: DataFrame.sample**. 2026. Disponível em: <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html>.

7. PINHEIRO, João Ismael Dantas et al. **Estatística básica: a arte de trabalhar com dados**. Rio de Janeiro: Elsevier, 2009.

8. STATISTICS CANADA. **Teacher’s Guide to Data Discovery**. Ottawa: Statistics Canada, 2010. Disponível em: <https://www150.statcan.gc.ca/n1/pub/12-593-x/12-593-x2010001-eng.pdf>.
